### ISSUE TITLE
AppVM menu shortcuts don't always start application when clicked

### ISSUE THREAD

**Issue by molnara** [QubesOS/qubes-issues#3473]
#### Qubes OS version:

Qubes release 4.0 (R4.0)

#### Affected TemplateVMs: 

fedora-26

---

### Steps to reproduce the behavior:

1. Click Qubes menu
2. Click Domain: personal -> personal: Text Editor
3. Repeat previous step many times, Click Domain: personal -> personal: Text Editor

### Expected behavior:

1. personal domain starts
2. Text Editor 1 should start
3. Text Editor 2, 3, etc should start

### Actual behavior:

I have a tough time reproducing this one, but it happens very frequently.  I'll click on say Text Editor and nothing happens.  I'll click it again and again nothing.  Usually It'll open on the 3rd or 4th attempt.  A few times I also had to restart the AppVM as I could not even get a Terminal to open.

### General notes:

Are there any type of logging for the menu items that I could investigate further?

From what I understand, these shortcuts are just executing `qvm-run` so I am not sure what is going on.

---

**Comment by Aekez** [comment]
I've encountered these too, both in Q3.2. and Q4.0. However, at least on my systems, it's happening much less on Q4.0., especially after RC-3 phase. To me it typically happens with the same app's or AppVM's. However, the below issue/fix might also contribute to why I see this issue rarely now.

Are you by chance using old Q3.2. templates or AppVM's? In the past I started out with my old Q3.2. AppVM's in Q4.0, and then systematically slowly replaced them all with a freshly made Q4.0 AppVM's and transfered all my files over to the new AppVM, and deleted the old 3.2. ones. While all the templates were of fresh Q4.0 origin, replacing all the 3.2. AppVM's has oddly enough made some significant improvements in speed and reliability. 

Maybe this may make difference to you as well?

**Comment by andrewdavidwong** [comment]
Possibly related: #2449.

**Comment by molnara** [comment]
Yes, I also experienced the same issue with 3.2.  I didn't migrate anything to 4.0, this is a completely fresh install.

**Comment by marmarek** [comment]
Looks like #3433. It may be also that you don't have enough memory to start new VM? Check `~/.xsession-errors` in dom0 - there you may find a relevant message. Currently GUI notifications about this case are broken...

**Comment by molnara** [comment]
I checked `~/.xsession-errors` in dom0 and nothing but a few warnings.

**Comment by DTMW2** [comment]
Can confirm that one, but with different behavior. I have a fedora-VM (no template, standalone) running on 4.0 which I imported from 3.2. All icons in the Q-menu for that VM never work. It is just not possible to start any application. I have to run a terminal from dom0 via qvm-run instead!?

What I tried to mitigate: remove the apps via "qubes settings -> Applications" and re-insert them. Yet no success.
